backend/src/achievements/dto/create-achievement.dto.ts

```diff
@@ -0,0 +1,7 @@
+import { PickType } from '@nestjs/swagger';
+import { AchievementDto } from './achievement.dto';
+
+export class CreateAchievementDto extends PickType(AchievementDto, [
+  'name',
+  'description',
+]) {}
```

Comment:
Perhaps the `icon` could be included as an optional field, so the user can set the icon at the same time they create the entity?

Reply:
@fyusuf-a told me we won't let the end user create new achievements, which means the current system is there for testing purposes mostly. In that regard, do you think i should allow the upload of a file, or maybe we could assume that if there is an icon for this achievement, its path uses its id or unique name and thus can be deduced rather than specified ? I don't know how much sense I'm making here but let me know what you think.

Reply:
Okay, that makes sense. I agree!